### PR TITLE
Events page ~ show tweets

### PR DESCRIPTION
**Pull request by mts**
# Summary

## Feature
- Implements lambda function [getRecentTweets()](https://eu-west-1.console.aws.amazon.com/lambda/home?region=eu-west-1#/functions/codestar-nl-serverless-test-getRecentTweets) to collect tweets and displays on events page
- Implements a number of UI components implemented using [Atomic Design Patterns](http://bradfrost.com/blog/post/atomic-web-design/)  for displaying a list of tweets. Atoms and molecules are used to implement the UI.

  - Atoms
     Atoms are the basic building blocks of matter. Applied to web interfaces, atoms are our HTML tags,      
     such as a form label, an input or a button.

     Atoms can also include more abstract elements like color palettes, fonts and even more invisible 
     aspects of an interface like animations.

     Like atoms in nature they’re fairly abstract and often not terribly useful on their own. However,      
     they’re good as a reference in the context of a pattern library as you can see all your global styles 
     laid out at a glance.

  - Molecules
     Things start getting more interesting and tangible when we start combining atoms together. 
     Molecules are groups of atoms bonded together and are the smallest fundamental units of a 
     compound. These molecules take on their own properties and serve as the backbone of our design 
     systems.

     For example, a form label, input or button aren’t too useful by themselves, but combine them 
     together as a form and now they can actually do something together.

     Building up to molecules from atoms encourages a “do one thing and do it well” mentality. While 
     molecules can be complex, as a rule of thumb they are relatively simple combinations of atoms built 
     for reuse.

## Enhancement
- Introduces [Sass Preprocessor integrated with CSS Modules](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#adding-a-css-preprocessor-sass-less-etc)
- Introduces [Storybook integrated with Typescript](https://github.com/storybooks/storybook) 
- Implements Storybook stories with [Knobs Addon](https://github.com/storybooks/storybook/tree/master/addons/knobs)
- Introduces [Enzyme](https://github.com/airbnb/enzyme) for UI `unit` and `snapshot` testing and extends `create-react-app` `Jest` 
  configuration (enhancement)

## Chore
- Introduces ESLint configuration for JavaScript files
- Changes `tabs` in code to `two spaces`

# Detail
- `.editorconfig`
  Replaces tabs to two spaces

- `.eslintrc`
  Configures ESLint for JavaScript files

- `.gitignore`
  Ignores `storybook-static` folder

- `.storybook/addons.js`
  Registers `Storybook` add-ons

- `.storybook/config.js`
  Configures `Storybook` to pick-up implemented stories

- `.storybook/webpack.config.js`
  Extends `Storybook` `Webpack` configuration for [Typescript](https://storybook.js.org/configurations/typescript-config/).

- `package.json`
   Extends packages and scripts

- `public/locales/en/events.json`
   `public/locales/nl/events.json`
   Add translations in English and Dutch languages

- `serverless.yml`
   `serverless/meetup.js`
   `serverless/util.js`
  Implements and configures `getRecentTweets()` lambda function

- `src/About/About.tsx`
  `src/Animations/AnimatedLogo.jsx`
  `src/Animations/DelayedFade.jsx`
  `src/Animations/LandscapeBackground.jsx`
  `src/AsyncComponent/AsyncComponent.jsx`
  `src/CaseHeader/CaseHeader.tsx`
  `src/Cases/Cases.tsx`
  `src/Cases/CasesList.ts`
  `src/Clients/Clients.jsx`
  `src/Contact/Contact.test.js`
  `src/Contact/Contact.tsx`
  `src/Container/Container.jsx`
  `src/EventCard/EventCard.jsx`
  `src/EventsHeader/DetailsSection.test.tsx`
  `src/EventsHeader/DetailsSection.tsx`
  `src/EventsHeader/EventsHeader.tsx`
  `src/EventsHeader/EventsHeaderButton.jsx`
  `src/EventsHeader/EventsHeaderMessage.tsx`
  `src/EventsHeader/HeaderContent.test.tsx`
  `src/EventsHeader/HeaderContent.tsx`
  `src/EventsHeader/constants.jsx`
  `src/Footer/Footer.jsx`
  `src/InlineLogo/InlineLogo.jsx`
  `src/InlineLogo/InlineLogo.test.jsx`
  `src/Intro/Intro.tsx`
  `src/JobDescription/JobDescription.tsx`
  `src/Jobs/JobCard.tsx`
  `src/Jobs/Jobs.tsx`
  `src/Jobs/JobsList.js`
  `src/Map/Map.jsx`
  `src/NavBar/EventsButton.test.tsx`
  `src/NavBar/EventsButton.tsx`
  `src/NavBar/NavBar.tsx`
  `src/NotFound/NotFound.jsx`
  `src/OurStack/OurStack.test.jsx`
  `src/OurStack/OurStack.tsx`
  `src/ResponsiveImage/ResponsiveImage.jsx`
  `src/Section/Section.jsx`
  `src/ShareButtons/ShareButtons.jsx`
  `src/ShareButtons/ShareButtons.test.tsx`
  `src/SideMenu/SideMenu.tsx`
  `src/typed-translate.ts`
   Adjusts tabs to two spaces  

- `src/NavBar/__snapshots__/EventsButton.test.tsx.snap`
   EventsButton snapshot test update.

- `src/components/Atoms/Action/Button/Button.module.css`
    `src/components/Atoms/Action/Button/Button.module.scss`
    `src/components/Atoms/Action/Button/Button.stories.js`
    `src/components/Atoms/Action/Button/Button.test.js`
    `src/components/Atoms/Action/Button/Button.tsx`
    `src/components/Atoms/Action/Button/__snapshots__/Button.test.js.snap`
    `src/components/Atoms/Action/Button/index.ts`
    Implements Button component

- `src/components/Atoms/Text/Heading/Heading.module.css`
     `src/components/Atoms/Text/Heading/Heading.module.scss`
     `src/components/Atoms/Text/Heading/Heading.stories.js`
     `src/components/Atoms/Text/Heading/Heading.test.js`
     `src/components/Atoms/Text/Heading/Heading.tsx`
     `src/components/Atoms/Text/Heading/__snapshots__/Heading.test.js.snap`
     `src/components/Atoms/Text/Heading/index.ts`
     Implements Heading component

- `src/components/Atoms/Text/Label/Label.module.css`
     `src/components/Atoms/Text/Label/Label.module.scss`
     `src/components/Atoms/Text/Label/Label.stories.js`
     `src/components/Atoms/Text/Label/Label.test.js`
     `src/components/Atoms/Text/Label/Label.tsx`
     `src/components/Atoms/Text/Label/__snapshots__/Label.test.js.snap`
     `src/components/Atoms/Text/Label/index.ts`
     Implements Label component

- `src/components/Events/Events.test.jsx (relocated)`
     `src/components/Events/Events.tsx (relocated)`
     `src/components/Molecules/List/Events/Events.module.css`
     `src/components/Molecules/List/Events/Events.module.scss`
     `src/components/Molecules/List/Events/Events.test.jsx`
     `src/components/Molecules/List/Events/Events.tsx`
     `src/components/Molecules/List/Events/index.ts`
     Adjusts Implements Events implementation adding TweetList component

- `src/components/Molecules/Card/Tweet/Tweet.module.css`
     `src/components/Molecules/Card/Tweet/Tweet.module.scss`
     `src/components/Molecules/Card/Tweet/Tweet.stories.js`
     `src/components/Molecules/Card/Tweet/Tweet.test.js`
     `src/components/Molecules/Card/Tweet/Tweet.tsx`
     `src/components/Molecules/Card/Tweet/__snapshots__/Tweet.test.js.snap`
     `src/components/Molecules/Card/Tweet/index.ts`
     Implements Tweet component

- `src/components/Molecules/List/TweetList/TweetList.stories.js`
     `src/components/Molecules/List/TweetList/TweetList.test.js`
     `src/components/Molecules/List/TweetList/TweetList.tsx`
     `src/components/Molecules/List/TweetList/__snapshots__/TweetList.test.js.snap`
     `src/components/Molecules/List/TweetList/index.ts`
     Implements TweetList component

- `src/eventsService.test.js`
     `src/eventsService.ts`
     `src/__tests__/eventsService/getCachedPastEvents.test.js`
     `src/__tests__/eventsService/getCachedRecentTweets.test.js`
     `src/__tests__/eventsService/getCachedUpcomingEvents.test.js`
     Extends eventsService implementation and test adding getCachedRecentTweets()

- `src/modules/EventsContainer/EventsContainer.jsx`
     Fetches data from getRecentTweets() lambda function through eventService

- `src/setupTests.js`
- `src/setupTests.ts`
   Extends `create-react-app` `jest` configuration introducing `Enzyme` and global function `renderToJSON` using `react-test-renderer` for snapshot tests

- `src/theming/config.module.css`
- `src/theming/config.module.scss`
   Configures `theming` in Sass

- `src/utility/enumeration.ts`
   Implements enumerations for UI components

- `tsconfig.json`
  Excludes `serverless` and `storybook-static` folders

# Newly introduced packages
- A simple javascript utility for conditionally joining classNames together
[classnames](https://github.com/JedWatson/classnames)
[@types/classnames](https://github.com/DefinitelyTyped/DefinitelyTyped#readme)

- A simple oauth API for node.js . This API allows users to authenticate against OAUTH providers, and thus act as OAuth consumers.
- Used when collecting tweets
[oauth](https://github.com/ciaranj/node-oauth/)
[defer-to-connect](https://github.com/szmarczak/defer-to-connect)

- JavaScript Testing utilities for React
[enzyme](https://github.com/airbnb/enzyme)
[enzyme-adapter-react-16](https://github.com/airbnb/enzyme/tree/master/packages/enzyme-adapter-react-16)
[@types/enzyme](https://github.com/DefinitelyTyped/DefinitelyTyped#readme)

- [Simplify tests with readable matchers](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#srcsetuptestsjs)
[jest-enzyme](https://github.com/FormidableLabs/enzyme-matchers/tree/master/packages/jest-enzyme)
[@types/jest](https://github.com/DefinitelyTyped/DefinitelyTyped#readme)
[ts-jest](https://github.com/kulshekhar/ts-jest)

- A thin wrapper around node-sass to replicate the --watch using chokidar instead of Gaze
[node-sass-chokidar](https://github.com/michaelwayman/node-sass-chokidar)
[npm-run-all](https://github.com/mysticatea/npm-run-all/blob/master/docs/run-p.md)

- Storybook

   - Storybook react
    [@storybook/react](https://github.com/storybooks/storybook/tree/master/app/react)

    - Add-ons
    [@storybook/addon-actions](https://github.com/storybooks/storybook/tree/master/addons/actions)
    [@storybook/addon-info](https://github.com/storybooks/storybook/tree/master/addons/info)
    [@storybook/addon-knobs](https://github.com/storybooks/storybook/tree/master/addons/knobs)
    [@storybook/addon-links](https://github.com/storybooks/storybook/tree/master/addons/links)
    [@storybook/addon-notes](https://github.com/storybooks/storybook/tree/master/addons/notes)
    [@storybook/addons](https://github.com/storybooks/storybook/tree/master/addons)
   - Use Storybook integrated with TypeScript
    [awesome-typescript-loader](https://github.com/s-panferov/awesome-typescript-loader)
    [react-docgen-typescript-webpack-plugin](https://github.com/strothj/react-docgen-typescript-loader)

   - Transpiling & linting
    [@babel/core](https://github.com/babel/babel/tree/master/packages/babel-core)
    [babel-eslint](https://github.com/babel/babel-eslint)
    [babel-loader](https://github.com/babel/babel-loader)

- Linting ES6

    [eslint](https://github.com/eslint/eslint)
    [eslint-config-react-app](https://github.com/facebook/create-react-app#readme)
    [eslint-plugin-import](https://github.com/gajus/eslint-plugin-flowtype)
    [eslint-plugin-jsx-a11y](https://github.com/evcohen/eslint-plugin-jsx-a11y#readme)
    [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react)

### CODE REVIEW

File: src/components/Atoms/Action/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import classNames from 'classnames/bind';
+import style from './Button.module.css';
+
+const cx = classNames.bind(style);
+
+interface IButtonProps {
+  variant: string;
+  children: any;
```
Comment (mvdam) [inline src/components/Atoms/Action/Button/Button.tsx]: `children` is of type `ReactNode` wich is exported from `React`

File: src/components/Atoms/Action/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+const Button = ({ variant, children }: IButtonProps) => {
+  return (
+    <button
+      className={cx('button', {
```
Comment (mvdam) [inline src/components/Atoms/Action/Button/Button.tsx]: I like this approach! 🥇

File: src/components/Atoms/Image/Icon/Icon.stories.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
```
Comment (mvdam) [inline src/components/Atoms/Image/Icon/Icon.stories.js]: Is it possible to use TypeScript files for this `*.stories.js` files? Then the TypeScript compiler will let you know when made (conflicting) changes to the components used in these files.

File: src/eventsService.ts [hunk 4]
```diff
@@ -1,44 +1,70 @@
...
-	if (process.env.REACT_APP_STAGE === 'dev') {
-		return `/mock/${lambdaName}.json`;
-	}
-	const AWS_PREFIX =
-		process.env.REACT_APP_STAGE === 'test' ? 'hjoutysc5k' : 'c3mmkmwyqi';
-	const AWS_STAGE = process.env.REACT_APP_STAGE === 'test' ? 'test' : 'prod';
-	return `https://${AWS_PREFIX}.execute-api.eu-west-1.amazonaws.com/${AWS_STAGE}/${lambdaName}`;
+  // if (process.env.REACT_APP_STAGE === 'dev') {
+  //   return `/mock/${lambdaName}.json`;
+  // }
```
Comment (mvdam) [inline src/eventsService.ts]: I think we want to enable this again?

File: src/eventsService.ts [hunk 5]
```diff
@@ -1,44 +1,70 @@
...
+
+  const AWS_PREFIX =
+    process.env.REACT_APP_STAGE === 'test' ||
+    process.env.REACT_APP_STAGE === 'dev'
```
Comment (mvdam) [inline src/eventsService.ts]: Can't we just use `'test'` and remove all the usages of `'dev'`?

File: src/fonts/selection.json [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+			"useClassSelector": true,
+			"color": 0,
+			"bgColor": 16777215,
+			"name": "icomoon",
+			"classSelector": ".icon"
+		},
+		"historySize": 50,
+		"gridSize": 16
+	}
+}
```
Comment (mvdam) [inline src/fonts/selection.json]: I think we already had some icon font (in Material UI?). Can't we just keep that one so we have less to maintain? (see https://material-ui.com/api/icon/). I prefer sticking to this `Icon` component so we don't need an extra dependency.

File: src/setupTests.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+import renderer from 'react-test-renderer';
+
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+global.renderToJSON = component => {
+  return renderer.create(component).toJSON();
+};
```
Comment (mvdam) [inline src/setupTests.ts]: Nice! 👍

File: src/theming/config.scss [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+$font-size-tiny: 8px;
+
+// colors
+$color-white: #fff;
+$color-grey: #c6c6c6;
+$color-black: #000;
+$color-blue: #00F;
+$color-event-blue: #8FC3EA;
+$color-red: #F00;
+$color-yellow: #FF0;
```
Comment (mvdam) [inline src/theming/config.scss]: 👍

File: serverless.yml [hunk 9]
```diff
@@ -78,6 +78,10 @@ package:
     - node_modules/to-readable-stream/**
     - node_modules/url-parse-lax/**
     - node_modules/wrappy/**
+    - node_modules/@szmarczak/**
```
[the remark this answers is not in the document]
Comment (mts) [reply serverless.yml]: it is a dependency of oauth used to call the twitter API `serverless/meetup.js`

File: src/components/Atoms/Action/Button/Button.tsx (hunk 1 above)
Comment (mts) [reply src/components/Atoms/Action/Button/Button.tsx]: I changed this :)

File: src/components/Atoms/Image/Icon/Icon.stories.js (hunk 3 above)
Comment (mts) [reply src/components/Atoms/Image/Icon/Icon.stories.js]: as discussed

File: src/eventsService.ts (hunk 5 above)
Comment (mts) [reply src/eventsService.ts]: as discussed

File: src/fonts/selection.json (hunk 6 above)
Comment (mts) [reply src/fonts/selection.json]: removed

File: src/eventsService.ts (hunk 4 above)
Comment (mts) [reply src/eventsService.ts]: As discussed. Enabled.

File: src/components/Atoms/Action/Button/Button.tsx (hunk 2 above)
Comment (mts) [reply src/components/Atoms/Action/Button/Button.tsx]: Me too 🥇

File: src/fonts/selection.json [hunk 10]
```diff
@@ -0,0 +1,261 @@
+{
```
Comment (mvdam) [inline src/fonts/selection.json]: This file can now be removed, right?

Comment (mts) [reply src/fonts/selection.json]: Removed.

**Comment by Tenchi2xh** [comment]
I made a mockup of the "alternative idea" from above by using the Chrome dev tools on the example from the docs:

### Our next Speaker:

image

**Comment by mvdam** [comment]
I really like the ` mockup of the "alternative idea"` from @Tenchi2xh! Really clean and simple :-)

**Comment by mdvanes** [comment]
I have accepted the changes. We created follow up tasks on Trello for some of the more complicated issues that are beginning to show with the growing of the application, e.g. splitting Serverless into a separate repo.
The UI elements that were mentioned in the last review are turned off for the moment, and will be done in a separate commit.

**Pull request merged**